WireDomain/Sources/WireDomain/Utilities/NonReentrantTaskManager.swift

```diff
@@ -0,0 +1,61 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+/// An object that facilitates non-reentrant calls to an async block.
+///
+/// Use this object to ensure that an async block is executed only once
+/// at any given time, regardless how many times the block is enqueued.
+/// Repeated invocations don't result in repeated executions.
```

Comment:
question: I am confused, does it not wait and when finish execute ?

Reply:
It does. What I wanted to say is that if the block takes 5 seconds and you call the method once every second, then the first invocation executes the block, the 2nd, 3rd, 4th, 5th invocations wait for the first block to end, then the 6th invocation executes a second block, and the remaining invocations wait for that secondly block to end.